File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/EventHandler.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.content.Context;
import android.support.annotation.NonNull;

/**
 * @author lukas
 * @since 05.07.18
 */
public interface EventHandler extends Proxy {
    int RUN_SCRIPT = 35;
    int RESTART = 28;
    int UNSET = 0;

    static EventHandler newInstance(@NonNull Context context, int action, String data) {
        try {
            return ProxyFactory.lightningProxy(context.getClassLoader()
                    .loadClass("net.pierrox.lightning_launcher.script.api.EventHandler")
                    .getConstructor(int.class, String.class)
                    .newInstance(action, data), EventHandler.class);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    int getAction();

    String getData();

    void setNext(EventHandler next);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/EventHandler.kt
package com.faendir.lightning_launcher.multitool.proxy

import android.content.Context

/**
 * @author lukas
 * @since 05.07.18
 */
interface EventHandler : Proxy {

    val action: Int

    val data: String?

    fun setNext(next: EventHandler?)

    companion object {
        const val RUN_SCRIPT = 35
        const val RESTART = 28
        const val UNSET = 0

        fun newInstance(context: Context, action: Int, data: String): EventHandler {
            try {
                return ProxyFactory.lightningProxy(context.classLoader
                        .loadClass("net.pierrox.lightning_launcher.script.api.EventHandler")
                        .getConstructor(Int::class.javaPrimitiveType, String::class.java)
                        .newInstance(action, data), EventHandler::class.java)
            } catch (e: Exception) {
                throw RuntimeException(e)
            }

        }
    }
}
